test/SampleImmutable.js

```diff
@@ -0,0 +1,24 @@
+const SampleImutable = artifacts.require('SampleImmutable');
+
+contract('Sample Immutable', accounts => {
+    it("Should deploy and retrieve the same value", async () => {
+        const value = 7;
+        const sample = await SampleImutable.new(value);
+        assert.equal(await sample.getImmutable(), value);
+    });
+
+    it("Should run sumToImmutable correctly", async () => {
+        const value = 10;
+        const sum = 55;
+        const sample = await SampleImutable.new(value);
+        assert.equal(await sample.sumToImmutable(), sum);
+    });
+
+    it("Should run storeAtImmutable correctly", async () => {
+        const value = 11;
```

Comment:
Let's use some larger number here just to amke sure we test different widths.
```suggestion
        const value = 114325;
```

Reply:
I've updated accordingly. At first I tried using a BN library but ended up stuck in some assertion errors,.Finally, I've used `Numbers.MAX_SAFE_INTEGER - 1`. 

Here are the bytecodes for the test.

bytecode with cons:
<code>
"0x60a060405234801561001057600080fd5b506040516102bf3803806102bf8339818101604052602081101561003357600080fd5b810190808051906020019092919050505080608081815250505060805161024461007b600039806101085280610154528061018f52806101c452806101ea52506102446000f3fe608060405234801561001057600080fd5b50600436106100575760003560e01c8063432acdf81461005c5780634c20a7331461007a5780634fd4f5de146100a85780638ff548f3146100c65780639dcedf6e146100e4575b600080fd5b610064610102565b6040518082815260200191505060405180910390f35b6100a66004803603602081101561009057600080fd5b810190808035906020019092919050505061014e565b005b6100b0610188565b6040518082815260200191505060405180910390f35b6100ce6101c2565b6040518082815260200191505060405180910390f35b6100ec6101e6565b6040518082815260200191505060405180910390f35b6000805b7f0000000000000000000000000000000000000000000000000000000000000000811161014a5780820191508082101561013f57600080fd5b600181019050610106565b5090565b806000807f000000000000000000000000000000000000000000000000000000000000000081526020019081526020016000208190555050565b60008060007f0000000000000000000000000000000000000000000000000000000000000000815260200190815260200160002054905090565b7f000000000000000000000000000000000000000000000000000000000000000081565b60007f000000000000000000000000000000000000000000000000000000000000000090509056fea2646970667358221220f9d7e6eb4a7f3cb5d71153973f22991713bdab11cea488eeced6031365ef6f6264736f6c634300060c0033000000000000000000000000000000000000000000000000001ffffffffffffe"
</code>
 - Here, the value of the immutable is stored in the final 64 bytes: `000000000000000000000000000000000000000000000000001ffffffffffffe`
 - All the immutable references are marked with a push32 and 64 zeros: `7f0000000000000000000000000000000000000000000000000000000000000000`

actual bytecode in the account:
 <code>
"0x608060405234801561001057600080fd5b50600436106100575760003560e01c8063432acdf81461005c5780634c20a7331461007a5780634fd4f5de146100a85780638ff548f3146100c65780639dcedf6e146100e4575b600080fd5b610064610102565b6040518082815260200191505060405180910390f35b6100a66004803603602081101561009057600080fd5b810190808035906020019092919050505061014e565b005b6100b0610188565b6040518082815260200191505060405180910390f35b6100ce6101c2565b6040518082815260200191505060405180910390f35b6100ec6101e6565b6040518082815260200191505060405180910390f35b6000805b7f000000000000000000000000000000000000000000000000001ffffffffffffe811161014a5780820191508082101561013f57600080fd5b600181019050610106565b5090565b806000807f000000000000000000000000000000000000000000000000001ffffffffffffe81526020019081526020016000208190555050565b60008060007f000000000000000000000000000000000000000000000000001ffffffffffffe815260200190815260200160002054905090565b7f000000000000000000000000000000000000000000000000001ffffffffffffe81565b60007f000000000000000000000000000000000000000000000000001ffffffffffffe90509056fea2646970667358221220f9d7e6eb4a7f3cb5d71153973f22991713bdab11cea488eeced6031365ef6f6264736f6c634300060c0033"
</code>
- all references now contain the value of the immutable